src/Panel/Plugins.php

```diff
@@ -75,9 +79,25 @@ public function read(string $type): string
 	{
 		$dist = [];
 
-		foreach ($this->files() as $file) {
+		$files = $this->files();
+
+		// filter out index.js files that have an index.dev.mjs counterpart (which takes precedence)
+		if ($type === 'js') {
+			$files = A::filter(
+				$files,
+				fn ($f) => !(
+					Str::endsWith($f, 'index.js') && F::exists(preg_replace('/\.js$/', '.dev.mjs', $f))
+				)
+			);
+		}
```

Comment:
I think this filtering might better done in `Plugins::files()`

Reply:
Wasn't sure here – the way it was implemented before, `::files()` just listed all *possible* plugin files and then `::read()` did the checks to see if they actually exist. That's why I kept `::files()` pure and added the additional filesystem access inside the `::read()` method instead of in `::files()`.

(but happy to change it if you want!)

Reply:
I tend to agree with Jonas. CSS and JS are not filtered in `Plugins::files()` either. On the other hand, the filtering logic would be much simpler in `files()` because we wouldn't need the regex.

Reply:
I see your arguments, I'd be fine to leave it where it is now. Can we just clean up the code Style a bit? We usually don't do this `!` to negate the result as it's harder to red and understand. Would be great to relive this

Reply:
Refactored the filter fn a little, what do you think?